### PR TITLE
Automate release process to Maven Central and Gradle Plugin Portal

### PR DESCRIPTION
**Pull request by nhachicha**
Fixes https://jira.mongodb.org/browse/RKOTLIN-709

### CODE REVIEW

**Review by rorbech — APPROVED** [review]
LGTM

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -1642,4 +1642,70 @@ jobs:
...
+    needs: [check-cache]
+    outputs:  
+      is_publish_build: ${{ steps.check_release.outputs.is_publish_build }}
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Get Git tag
+        run: |
+          gitTag=$(git describe --tags --exact-match HEAD) || echo "NONE"
```
Comment (clementetb) [inline .github/workflows/pr.yml:1659]: If the tag does not exist the tag would be "NONE", shouldn't we stop the release if not gittag is defined?

**Review by clementetb — COMMENTED** [review]
Looks good except for that gittag comment.

**Review by clementetb — COMMENTED** [review]
Looks good except for that gittag comment.

File: .github/workflows/pr.yml (hunk 1 above)
Comment (nhachicha) [reply .github/workflows/pr.yml:1659]: If the gittag is `NONE` `is_publish_build` will be `false` in the next step which will skip the release 
image

**Pull request merged**